api/src/main/java/org/apache/iceberg/actions/DeleteOrphanFiles.java

```diff
@@ -80,9 +84,16 @@ public interface DeleteOrphanFiles extends Action<DeleteOrphanFiles, DeleteOrpha
    *
    * @param executorService the service to use
    * @return this for method chaining
+   * @deprecated All deletes should be performed using the bulk delete api if available. Use FileIO
+   *     specific parallelism controls to adjust bulk delete concurrency within that api.
    */
+  @Deprecated
```

Comment:
I am not sure the deprecation is worth it. Instead of deprecating a bunch of widely used methods in actions and arguments in procedures, can we consider supporting external executor services with bulk deletes so that these arguments can be still supported?

I like the idea of bulk deletes in `HadoopFileIO`, but I don't see anything fundamentally bad in an ability to override the delete parallelism in actions or procedures. Especially, because it can make the implementation simpler. We already discussed passing executor services in the original PR that added the bulk API.

Reply:
I really don't think we should support external executor services. I think we are just pushing a ton of complexity into the user space which is only wildly used because our defaults are not good. If we don't allow passing an external executor to S3FileIO (or file io's in general) I don't think we should allow it here either.

Reply:
The original proposal for `SupportsBulkOperations` included a way to pass a custom executor service, which we removed until we had a use case. Without having a default executor service in file IO, we would be pushing the complexity on the user. However, I was talking about having a default executor service in all file IO implementations but allowing to pass a custom one in `SupportsBulkOperations`. That way, our defaults are always good but if needed, one may pass a custom executor service.

```
void deleteFiles(Iterable<String> pathsToDelete) throws BulkDeletionFailureException;
void deleteFiles(ExecutorService executorService, Iterable<String> pathsToDelete) throws BulkDeletionFailureException;
```